### PR TITLE
Batch Dependabot PRs

### PR DESCRIPTION
**Pull request by rosstimothy**
Leverages https://github.com/Legal-and-General/dependabot-batcher to create and/or maintain a merged PR (and branch) of all the Dependabot PRs and close them as they are merged in, leaving a single Dependabot PR behind.

The Dependabot schedule is set for Sunday at 9:00am and the batcher workflow is configured to run Sunday at 9:00pm.

All workflows that run on pull requests open have been configured to ignore branches prefixed with `dependabot/`. Since the dependabot PRs are all going to be closed and batched there is no need to waste CI minutes on them. Any new workflows will require the same check to prevent running on dependabot PRs.

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -54,6 +62,8 @@ updates:
     directory: "/"
     schedule:
       interval: weekly
+      day: "sunday"
```
Comment (codingllama) [inline .github/dependabot.yml:65]: Set cargo updates to sunday/9am too?

File: .github/dependabot.yml [hunk 2]
```diff
@@ -29,6 +31,10 @@ updates:
       - dependency-name: github.com/vulcand/predicate
       # TODO(tross): Update after Firestore is updated
       - dependency-name: google.golang.org/genproto
+      # TODO(tross): Update after Firestore is updated
+      - dependency-name: cloud.google.com/go/iam
+      # TODO(tross): Update after /x/crypto is updated
+      - dependency-name: github.com/go-webauthn/webauthn
```
Comment (codingllama) [inline .github/dependabot.yml:37]: Could we let these run, but rely on previous ignores/manual changes to rollback crypto?

I suppose #19579 is making this a moot point anyway.

File: .github/workflows/dependabot-batcher.yaml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+# Dependabot PRs. The schedule is set to exactly 12 hours after the
+# Dependabot schedule.
+#
+# All workflows that run on PR should be skipped for Dependabot PRs
+# to prevent wasting CI cycles.
+name: 'Dependabot Batcher'
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 21 * * 0' # At 9:00 PM every Sunday
```
Comment (codingllama) [inline .github/workflows/dependabot-batcher.yaml:12]:
```suggestion
    - cron: '0 21 * * 0' # At 9:00 PM every Sunday UTC
```

(assuming it is UTC)

**Review by codingllama — APPROVED** [review]
Thanks for putting this together. LGTM.

**Review by zmb3 — APPROVED** [review]

**Review by codingllama — APPROVED** [review]

**Comment by pauleustice** [comment]
@rosstimothy I'm one of the creators of the Dependabot Batcher. Please let us know how you get on with it, and if you have any suggestions!

**Pull request merged**